### ISSUE TITLE
brave-browser launches the most recent install (.deb packages)

### ISSUE THREAD

**Issue by srirambv** [brave/brave-browser#1324]
## Description 

brave-browser launches the most recent install 

## Steps to Reproduce

   1. Install Brave-Browser-Dev on Linux
   2. Run `brave-browser` from terminal launches Dev build
   3. Install Brave-Browser-Beta after installing Brave-Browser-Dev
   4. Run `brave-browser` from terminal launches Beta build
   5. Uninstall Beta from terminal shows updating alternatives which starts to point `brave-browser` to `brave-browser-dev`
 ![image](https://user-images.githubusercontent.com/17010094/46170259-4766aa80-c26b-11e8-9b80-6d96405732ee.png)

## Actual result:

`brave-browser` launches the most recent install of Brave(Dev/Beta)

## Expected result:
`brave-browser` should not launch Beta/Dev build, should be specified only for Release build. Only `brave-browser-dev` should launch Dev build and `brave-browser-beta` should launch Beta build

## Reproduces how often: 

Easy

## Brave version (chrome://version info)

All Linux version

### Reproducible on current release:

N/A for b-l 

### Website problems only:  
- Does the issue resolve itself when disabling Brave Shields? 
- Is the issue reproducible on the latest version of Chrome?

**Comment by srirambv** [comment]
+1 from @Techguyprivate via #2281

**Comment by Techguyprivate** [comment]
Please solve the problem . It is now over a month . I am able to use brave stable for the last 3 days cause brave stable got recently updated. So I am able to open it. Otherwise , if a new beta/dev version will get published & updated , I will be unable to use brave stable . I didn't able to use my Brave stable for 20 days , where all my passwords, bookmarks are stored.

**Comment by srirambv** [comment]
++ on uplifting this. Gets really hard for testing as well. 

cc: @rebron @kjozwiak @bbondy

**Comment by Techguyprivate** [comment]
Meantime, any tweaks, workarounds, solution, clever or dirty hacks. Thank you.

**Comment by srirambv** [comment]
@Techguyprivate  the only work around(sorry for this) I can suggest for now is to uninstall and reinstall the release version once beta gets updated via `apt update`. This shortcut issue only affects release and beta channels.

**Comment by Techguyprivate** [comment]
I didn't understand. I should wait for a beta update ? After updating the beta , only and only then , I should uninstall, and reinstall release. If I uninstall the  stable normal way, not purge or anything else  in Linux mint  , will I lose my tons of bookmarks, passwords, history? Thank you in advance anyway for such a product & service.

**Comment by srirambv** [comment]
@Techguyprivate I think i put the workaround in a wrong way. Here's what you can do to actually use both release/beta without having to purge your profile. If you install a new beta build, then release and beta shortcut will launch only beta so you can just run the following steps to retain your release profile
```
sudo apt remove brave-browser
sudo apt install brave-browser
```
☝️ this will just remove the release version that is installed and reinstall the latest one keeping your profile intact.

**Comment by Techguyprivate** [comment]
Solved for Windows and ubuntu without uninstalling anything. 

Not solved for Linux mint 19.1 .

**Comment by fmarier** [comment]
Here's my work-around for this on Ubuntu 18.04:

```
cd /etc/alternatives/
rm brave-browser
ln -s /opt/brave.com/brave/brave-browser .
```

After this, launching `brave-browser` will launch the release browser as expected.

The problem is twofold:
1. Both the beta and dev packages divert the `/usr/bin/brave-browser` binary.
2. The `brave-browser` package doesn't register its binary as an alternative for the `/usr/bin/brave-browser` binary.

You can see that second point by doing:

```
$ update-alternatives --config brave-browser
There are 2 choices for the alternative brave-browser (providing /usr/bin/brave-browser).

  Selection    Path                         Priority   Status
------------------------------------------------------------
  0            /usr/bin/brave-browser-beta   150       auto mode
  1            /usr/bin/brave-browser-beta   150       manual mode
  2            /usr/bin/brave-browser-dev    0         manual mode

Press <enter> to keep the current choice[*], or type selection number: 
```
I do agree with the *expected result* from the bug description. We should remove the diversion from the `-beta` and `-dev` packages.

**Comment by riastradh-brave** [comment]
No opinion on whether `/usr/bin/brave-browser` should be a symlink to an alternative, but at least the release, beta, and dev packages should agree on whether the `/usr/bin/brave-browser` file _is_ a symlink to an alternative -- currently the beta and dev packages think it is, and update it with `update-alternatives`, while the release package thinks it isn't, so it is not presented as an alternative.

In other words, either the following fragments should be _unconditional_ on the release channel, or they shouldn't happen at all:

```
# brave-browser.postinst
if [ '"brave-browser"' != '"brave-browser"' ]; then
  update-alternatives --install /usr/bin/brave-browser brave-browser \
    /usr/bin/brave-browser $PRIORITY
fi

# brave-browser-beta.postinst
if [ '"brave-browser"' != '"brave-browser-beta"' ]; then
  update-alternatives --install /usr/bin/brave-browser brave-browser \
    /usr/bin/brave-browser-beta $PRIORITY
fi

# brave-browser-dev.postinst
if [ '"brave-browser"' != '"brave-browser-dev"' ]; then
  update-alternatives --install /usr/bin/brave-browser brave-browser \
    /usr/bin/brave-browser-dev $PRIORITY
fi
```

**Comment by Techguyprivate** [comment]
Today, Clicking on brave stable , open up brave beta on Ubuntu & Linux mint . I did remove brace , then installed it as suggested above. But the problem still occurs.

**Comment by fmarier** [comment]
@Techguyprivate did you try the work-around [mentioned above](https://github.com/brave/brave-browser/issues/1324#issuecomment-454611462)?

As the `root` user:
```
cd /etc/alternatives/
rm brave-browser
ln -s /opt/brave.com/brave/brave-browser .
```

**Comment by Techguyprivate** [comment]
@fmarier  It solved the problem. But  I can confirm that I did the following in my various linux os few months ago.

"  sudo apt remove brave-browser
sudo apt install brave-browser " as suggested by . 

But the problem reappears again & again. As far as I can remember, recently the stable release got updated to chromium 72.121 something( for the file api security bug ) and then to chromium  73 . After that updating the beta releases , the problem reappears. 

For some more information, I have 3 release channels installed, cause I want to test , learn, get acquainted with the BAT system & ad systems.  I recently installed nightly releases even. But on my ubuntu  , I have no nightly release installed cause I saw the nightly releases recently, but the problem persists.

**Comment by Techguyprivate** [comment]
@fmarier  Will changing the directory or removing brave browser , remove my profile?  

The problem is little frustrating cause I have lot of sites with already logged in , in brave stable like fb, twitter,more importantly  bank sites & other numerous sites.  Whenever I log into any sites or any sites that require my password, I give much attention to the url for phishing attacks. Also I use long passwords , so typing long passwords again & again is frustrating, even if I use keepass . 

The problem is that I am facing this issue for the last 6 months continously when brave stable or beta updated.   I don't get access to the brave stable once brave beta updated. Then clicking on brave stable opens up brave beta. SO I get locked out of brave stable. I can't use brave beta cause it is beta. so.

**Comment by fmarier** [comment]
Yes, that work-around is only temporary. As soon as `brave-browser-beta` or `brave-browser-dev` gets updated, the symlink needs to be fixed again.

You're right, it's really annoying. I have the same problem on my laptop. I will try to get to it soon.

**Comment by kjozwiak** [comment]
@rebron we need to revisit this and reprioritize to either `P3` or `P2`. We're going to have more users on linux running into the above the more we promote the different channels.

**Comment by Techguyprivate** [comment]
Just want to state  that today brave stable got updated to 61.52 (with an  Upgraded Chromium to 73.0.3683.86) , the problem reappeared again. Just yesterday, I ran the following. 

 cd /etc/alternatives/
rm brave-browser
ln -s /opt/brave.com/brave/brave-browser .

Quiet a irritating problem. Again & again , I am accidentally opening up Brave beta ( although I am clicking on Brave stable) . Happened at least 20 times in 6months. Now-a-days , I forced to go to about page to see whether it is brave stable  or beta.

But anyway, thank you for the awesome product.

**Comment by fmarier** [comment]
A simpler fix than [the one I suggested previously](https://github.com/brave/brave-browser/issues/1324#issuecomment-454611462) is to simply rename the binary shipped by the release package in `/usr/bin/` to `/usr/bin/brave-browser-stable` and let that get installed as the highest-priority alternative for the `/usr/bin/brave-browser` binary.

That's what the Fedora packages do already and so this is only a regression on Debian-based systems since brave/brave-core#45 (specifically https://github.com/brave/brave-core/pull/45/commits/c693f8901139957f6395d9087d23ac04191e69a5) got merged.

**Comment by Techguyprivate** [comment]
Again, opened up brave beta when clicking on brave stable. Is it solved ? Or do I need to do anything else other than the things mentioned above? 

Thank you in advanced anyway.

**Comment by fmarier** [comment]
It's solved in version 0.66, which isn't released yet (the current one is 0.63). Until that version comes out, you will, unfortunately, have to use the steps from https://github.com/brave/brave-browser/issues/1324#issuecomment-474936837.

I appreciate your patience with this. My apologies for the time it took to get this resolved.

**Comment by srirambv** [comment]
Verification passed by @GeetaSarvadnya on 

Brave | 0.66.98 Chromium: 75.0.3770.100 (Official Build) beta(64-bit)
-- | --
Revision | cd0b15c8b6a4e70c44e27f35c37a4029bad3e3b0-refs/branch-heads/3770@{#1033}
OS | Linux
